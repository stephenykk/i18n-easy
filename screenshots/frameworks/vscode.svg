<?xml version="1.0" encoding="UTF-8"?>
<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="17pt" height="16pt" viewBox="0 0 17 16" version="1.1">
<g id="surface1">
<path style=" stroke:none;fill-rule:nonzero;fill:rgb(14.117647%,53.72549%,79.215686%);fill-opacity:1;" d="M 0.492188 5.613281 C 0.492188 5.613281 0.0820312 5.347656 0.570312 4.976562 L 1.703125 4.035156 C 1.703125 4.035156 2.03125 3.714844 2.375 3.988281 L 12.839844 11.390625 L 12.839844 14.9375 C 12.839844 14.9375 12.832031 15.496094 12.070312 15.433594 Z M 0.492188 5.613281 "/>
<path style=" stroke:none;fill-rule:nonzero;fill:rgb(6.27451%,43.921569%,70.196078%);fill-opacity:1;" d="M 3.1875 7.90625 L 0.492188 10.195312 C 0.492188 10.195312 0.210938 10.386719 0.492188 10.730469 L 1.742188 11.796875 C 1.742188 11.796875 2.042969 12.09375 2.480469 11.753906 L 5.339844 9.730469 Z M 3.1875 7.90625 "/>
<path style=" stroke:none;fill-rule:nonzero;fill:rgb(3.137255%,46.666667%,72.54902%);fill-opacity:1;" d="M 7.925781 7.925781 L 12.871094 4.398438 L 12.839844 0.863281 C 12.839844 0.863281 12.628906 0.09375 11.925781 0.496094 L 5.339844 6.09375 Z M 7.925781 7.925781 "/>
<path style=" stroke:none;fill-rule:nonzero;fill:rgb(23.529412%,60%,83.137255%);fill-opacity:1;" d="M 12.070312 15.441406 C 12.355469 15.714844 12.699219 15.625 12.699219 15.625 L 16.558594 13.851562 C 17.054688 13.539062 16.984375 13.148438 16.984375 13.148438 L 16.984375 2.550781 C 16.984375 2.085938 16.472656 1.925781 16.472656 1.925781 L 13.132812 0.425781 C 12.402344 0 11.925781 0.496094 11.925781 0.496094 C 11.925781 0.496094 12.539062 0.0820312 12.839844 0.863281 L 12.839844 14.878906 C 12.839844 14.972656 12.816406 15.066406 12.773438 15.152344 C 12.683594 15.3125 12.496094 15.46875 12.035156 15.40625 Z M 12.070312 15.441406 "/>
</g>
</svg>
