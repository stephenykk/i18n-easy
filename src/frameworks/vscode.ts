import { basename } from 'path'
import { TextDocument } from 'vscode'
import { Framework } from './base'
import { LanguageId } from '~/utils'

class VSCodeFramework extends Framework {
  id = 'vscode'
  display = 'VS Code'

  detection = {
    packageJSON: [
      'vscode',
      '@types/vscode',
    ],
  }

  enabledParsers = ['json']

  languageIds: LanguageId[] = [
    'json',
    'javascript',
    'typescript',
  ]

  pathMatcher = () => 'package.nls.?{locale?}.json'

  usageMatchRegex = (languageIds?: string, filename?: string) => {
    if (filename && basename(filename) === 'package.json')
      return '"%({key})%"'
    // for visualize the regex, you can use https://regexper.com/
    return '(?:i18n[ (]path=|v-t=[\'"`{]|(?:this\\.|\\$|i18n\\.)(?:(?:d|n)\\(.*?,|(?:t|tc|te)\\()\\s*)[\'"`]({key})[\'"`]'
  }

  refactorTemplates(keypath: string, args?: string[], doc?: TextDocument) {
    if (doc?.languageId === 'json') {
      return [
        `%${keypath}%`,
        keypath,
      ]
    }

    return [
      `i18n.t('${keypath}')`,
      keypath,
    ]
  }
}

export default VSCodeFramework
