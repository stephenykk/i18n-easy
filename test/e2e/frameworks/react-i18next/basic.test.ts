import { window } from 'vscode'
import { openFile, Global, is, not, expect, timeout, setupTest, getExt, KeyDetector } from '../../ctx'

setupTest('React with i18next', () => {
  it('opens entry file', async() => {
    await openFile('package.json')
  })

  it('is active', () => {
    const ext = getExt()
    is(ext?.isActive, true)
  })

  it('enables correct frameworks', async() => {
    not(Global, undefined)
    is(Global.enabled, true)
    is(Global.enabledFrameworks.length, 3)
    is(Global.enabledFrameworks[0].id, 'react')
    is(Global.enabledFrameworks[1].id, 'i18next')
    is(Global.enabledFrameworks[2].id, 'general')
  })

  it('get correct coverage report', async() => {
    await timeout(500)
    not(Global, undefined)
    not(Global.loader, undefined)
    const coverage = Global.loader.getCoverage('en')
    expect(coverage).to.matchSnapshot()
  })

  it('get keys', async() => {
    await openFile('src/App.jsx')
    await timeout(500)
    const keys = KeyDetector.getKeys(window.activeTextEditor!.document)
    expect(keys).to.matchSnapshot()
  })
})
