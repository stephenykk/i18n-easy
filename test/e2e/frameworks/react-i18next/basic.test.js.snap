// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React with i18next get correct coverage report 1`] = `
Object {
  "allKeys": Array [
    "pages.home.title",
    "translation.description.part1",
    "translation.description.part2",
    "translation.foo.bar",
    "translation.title",
    "translation.titlew",
  ],
  "emptyKeys": Array [],
  "locale": "en",
  "missing": 0,
  "missingKeys": Array [],
  "total": 6,
  "translated": 6,
  "translatedKeys": Array [
    "pages.home.title",
    "translation.description.part1",
    "translation.description.part2",
    "translation.foo.bar",
    "translation.title",
    "translation.titlew",
  ],
}
`;

exports[`React with i18next get keys 1`] = `
Array [
  Object {
    "end": 553,
    "key": "translation.title",
    "quoted": true,
    "start": 536,
  },
  Object {
    "end": 617,
    "key": "translation.title",
    "quoted": true,
    "start": 600,
  },
  Object {
    "end": 684,
    "key": "translation.title",
    "quoted": true,
    "start": 667,
  },
  Object {
    "end": 899,
    "key": "translation.description.part1",
    "quoted": true,
    "start": 870,
  },
  Object {
    "end": 1493,
    "key": "translation.description.part2",
    "quoted": true,
    "start": 1464,
  },
  Object {
    "end": 1579,
    "key": "translation.description.part2",
    "quoted": false,
    "start": 1550,
  },
  Object {
    "end": 1747,
    "key": "translation.foo.bar",
    "quoted": true,
    "start": 1744,
  },
  Object {
    "end": 1796,
    "key": "pages.home.title",
    "quoted": true,
    "start": 1780,
  },
  Object {
    "end": 1820,
    "key": "pages.home.title",
    "quoted": true,
    "start": 1804,
  },
  Object {
    "end": 1928,
    "key": "pages.home.title",
    "quoted": true,
    "start": 1923,
  },
  Object {
    "end": 1978,
    "key": "translation.title",
    "quoted": true,
    "start": 1961,
  },
]
`;
